plugins/de.cau.cs.kieler.klighd.lsp/src/de/cau/cs/kieler/klighd/lsp/KGraphDiagramUpdater.xtend

```diff
@@ -0,0 +1,280 @@
+     * @param model The new model that should be shown for the server.
+     * @param key The key to access the diagram state maps.
+     */
+    synchronized def void prepareModel(KGraphDiagramServer server, Object model, String key) {
+
+        val properties = new KlighdSynthesisProperties()
+        var SprottyViewer viewer = null
+        var String synthesisId
+        synchronized (diagramState) {
+            val iViewer = diagramState.getViewer()
+            if (iViewer instanceof SprottyViewer) {
+                viewer = iViewer
+            }
+
+            synthesisId = diagramState.getSynthesisId(key)
+        }
+
+        // Set properties.
+        if (synthesisId !== null) {
+            // If the synthesisId is null, KLighD will use the a default synthesis defined for this model.
+            properties.setProperty(KlighdSynthesisProperties.REQUESTED_DIAGRAM_SYNTHESIS, synthesisId)
+        }
+
+        // Save previous synthesis options to restore later.
+        storeCurrentSynthesisOptions()
+        // Configure options.
+        var Map<SynthesisOption, Object> recentSynthesisOptions = null
+        synchronized (diagramState) {
+            recentSynthesisOptions = diagramState.recentSynthesisOptions
+        }
+        properties.configureSynthesisOptionValues(recentSynthesisOptions)
+
+        // Indicated if the model type changed against the current model
+        var modelTypeChanged = false
+        var ViewContext viewContext = null
+
+        if (viewer === null || viewer.viewContext === null) {
+            // if viewer or context does not exist always init view
+            modelTypeChanged = true
+        } else {
+            viewContext = viewer.viewContext
+            if (viewContext.inputModel === null || viewContext.inputModel.class !== model.class) {
+                modelTypeChanged = true
+            }
+            if (!KlighdDataManager.instance.getSynthesisID(viewContext.getDiagramSynthesis()).equals(synthesisId)) {
+                // In case the synthesis changed the sidebar should be updated
+                modelTypeChanged = true
+            }
+        }
+
+        // If the type changed the view must be reinitialized to provide a correct ViewContext
+        // otherwise the ViewContext can be simply updated.
+        if (modelTypeChanged) {
+            // Configure the ViewContext and the KlighD synthesis to generate the KGraph model correctly.
+            properties.useViewer(SprottyViewer.ID)
+            // needs to be a IDiagramWorkbenchPart, as it calls the standard constructor.
+            // TODO: The ViewContext should have a default constructor for non-SWT-based viewer.
+            viewContext = new ViewContext(null as IDiagramWorkbenchPart, model).configure(properties)
+            viewer = viewContext.createViewer(null, null) as SprottyViewer
+            viewer.diagramServer = server as KGraphDiagramServer
+            viewer.viewContext = viewContext
```

Comment:
I suggest to add a ticket for this.

Reply:
To fix this cleanly, this would include to remove Eclipse.ui dependencies from all the KLighD interfaces and dig deep into KLighD's code to do so. As discussed for the De-Eclipsification of KLighD, this currently is a 'won't do'.